packages/abi-typegen/test/fixtures/templates/predicate-with-configurable/main.hbs

```diff
@@ -95,18 +101,18 @@ const abi = {
 
 const bytecode = '0x000';
 
-export class MyPredicate extends Predicate<MyPredicateInputs> {
+export class MyPredicate extends Predicate<MyPredicateInputs, MyPredicateConfigurables> {
 
   static readonly abi = abi;
   static readonly bytecode = bytecode;
 
-  constructor(provider: Provider, predicateData?: MyPredicateInputs, configurables?: MyPredicateConfigurables) {
+  constructor(params: MyPredicateParameters) {
     super({
       abi,
       bytecode,
-      provider,
-      inputData: predicateData,
-      configurableConstants: configurables,
+      provider: params.provider,
+      data: params?.data,
+      configurableConstants: params?.configurableConstants,
```

Comment:
Do we need the `?`, considering `params` are not optional?

```suggestion
      data: params.data,
      configurableConstants: params.configurableConstants,
```

Reply:
c51c54715898ef288df16082a0ec70ae4563eb87